lib/fs/basicfs_test.go

```diff
@@ -365,16 +365,16 @@ func TestRooted(t *testing.T) {
 		{"baz/foo/", "/bar/baz", "baz/foo/bar/baz", true},
 
 		// Not escape attempts, but oddly formatted relative paths. Disallowed.
-		{"foo", "./bar", "", false},
-		{"baz/foo", "./bar", "", false},
-		{"foo", "./bar/baz", "", false},
-		{"baz/foo", "./bar/baz", "", false},
-		{"baz/foo", "bar/../baz", "", false},
-		{"baz/foo", "/bar/../baz", "", false},
-		{"baz/foo", "./bar/../baz", "", false},
-		{"baz/foo", "bar/../baz", "", false},
-		{"baz/foo", "/bar/../baz", "", false},
-		{"baz/foo", "./bar/../baz", "", false},
+		{"foo", "", "foo/", true},
```

Comment:
Canonicalize now returns stuff ending in slashes?

Reply:
`rooted` does and already did before this PR for the root path of the fs (and only the root). That's a side effect of needing to fulfill the prefix test (`if !strings.HasPrefix(joined, expectedPrefix)`). What I changed is that `"/"` and `""` are accepted as the same as `"."`. Paths like `/foo` were already allowed, so it seems just consistent to allow this as well (or disallow "/foo").